# ==============================================================================
# FIXING COMMIT DIFF
# ==============================================================================
# File: docshell/base/BrowsingContext.h
# Commit: 7333d4f4c7d1
# Full Hash: 7333d4f4c7d1b5ad585d222157067aa4276f9945
# Author: Nika Layzell <nika@thelayzells.com>
# Date: 2019-09-10 03:29:56
# Description:
#   Bug 1579437 - Check for discarded BrowsingContext in more places, r=farre
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D45088
# ==============================================================================

diff -r 941fe2efca5d -r 7333d4f4c7d1 docshell/base/BrowsingContext.h
--- a/docshell/base/BrowsingContext.h	Mon Sep 09 17:30:08 2019 +0000
+++ b/docshell/base/BrowsingContext.h	Mon Sep 09 12:04:13 2019 +0000
@@ -357,8 +357,10 @@
     // in all processes. This method will call the correct `MaySet` and
     // `DidSet` methods, as well as move the value.
     //
+    // If the target has been discarded, changes will be ignored.
+    //
     // NOTE: This method mutates `this`, resetting all members to `Nothing()`
-    void Commit(BrowsingContext* aOwner);
+    nsresult Commit(BrowsingContext* aOwner);
 
     // This method should be called before invoking `Apply` on this transaction
     // object.